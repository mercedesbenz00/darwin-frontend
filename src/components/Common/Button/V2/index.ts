export * from './_svg'

export { default as CustomButton } from './CustomButton/CustomButton.vue'
export { default as IconButton } from './IconButton/IconButton.vue'
export {
  default as RoundedTriToggleButton
} from './RoundedTriToggleButton/RoundedTriToggleButton.vue'

export * from './types'
