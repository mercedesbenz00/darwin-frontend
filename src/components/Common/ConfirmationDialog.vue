<template>
  <modal
    :name="name"
    width="30%"
    height="auto"
    :min-width="200"
    :max-width="500"
    transition="pop-out"
    :classes="['confirmation-dialog']"
    :click-to-close="false"
    @before-open="onBeforeOpen"
    @closed="onClosed"
  >
    <div class="modal__header">
      <div
        class="modal__header__close"
        @click="cancel"
      >
        &#10005;
      </div>
    </div>
    <div class="modal__content">
      <div
        v-if="$slots.title"
        class="confirm__title"
      >
        <slot name="title" />
      </div>
      <p
        v-else
        class="confirm__title"
      >
        {{ title }}
      </p>
      <div
        v-if="$slots.detail"
        class="confirm__text"
      >
        <slot name="detail" />
      </div>
      <p
        v-else
        class="confirm__text"
        v-html="detail"
      />
    </div>
    <div class="modal__footer">
      <slot
        name="footer"
      >
        <secondary-button
          class="modal__footer-left"
          @click="cancel"
        >
          {{ cancelText }}
        </secondary-button>
        <positive-button
          class="modal__footer-right"
          :disabled="loading"
          @click="confirm"
        >
          {{ confirmText }}
        </positive-button>
      </slot>
    </div>
  </modal>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

/**
 * Wrapper around a vue-js-modal
 *
 * The name property is used to open and close the modal,
 * so multiple modals rendered within the same template need to have unique names.
 */

@Component({
  name: 'confirmation-dialog'
})
export default class ConfirmationDialog extends Vue {
  @Prop({
    required: false,
    type: String as () => string,
    default: 'Are you sure?'
  })
  title!: string

  /**
   * The raw HTML to show in the center of the dialog.
   *
   * This is an unsafe way to render HTML and should be gradually outphased in
   * favor of the #detail slot.
   */
  @Prop({
    required: false,
    type: String as () => string,
    default: 'Are you sure?'
  })
  detail!: string

  @Prop({ required: true })
  name!: string

  @Prop({ required: false, default: 'Ok' })
  confirmText!: string

  @Prop({ required: false, default: 'Cancel' })
  cancelText!: string

  @Prop({ required: false, default: false, type: Boolean })
  loading!: boolean

  /**
   * Ugly and hacky, but no better idea at the moment.
   * The default modal behavior is to put the main app sidebar into the
   * background on open, and bring it back to front on close.
   *
   * When the modal is declared in the sidebar, this causes an issue where it
   * cannot be shown, because it gets obscured by the main content area.
   *
   * In such cases, we set this prop to `true`, in order to prevent that default
   * behavior.
   */
  @Prop({ required: false, default: false, type: Boolean as () => boolean })
  sidebar!: boolean

  data: any = null

  onBeforeOpen () {
    if (this.sidebar) { return }
    this.$store.dispatch('ui/putBackSidebar')
  }

  onClosed () {
    if (this.sidebar) { return }
    this.$store.dispatch('ui/bringFrontSidebar')
  }

  show (data?: any) {
    this.data = data || null
    this.$modal.show(this.name)
  }

  cancel () {
    if (this.data) {
      this.$emit('canceled', this.data)
    } else {
      this.$emit('canceled')
    }
    this.close()
  }

  close () {
    this.data = null
    this.$modal.hide(this.name)
  }

  confirm () {
    if (this.data) {
      this.$emit('confirmed', this.data)
    } else {
      this.$emit('confirmed')
    }
  }
}
</script>

<!-- eslint-disable vue-scoped-css/enforce-style-type -->
<style lang="scss">
.confirmation-dialog {
  display: flex;
  flex-direction: column;

  &.v--modal-box {
    box-shadow: 0px 15px 30px rgba(33, 64, 92, .3);
  }
}
</style>

<style lang="scss" scoped>
.confirmation-dialog {
  .modal__header {
    background: $colorLineGrey;
  }

  .modal__content {
    padding: 0 30px;
    background: $colorLineGrey;
    max-height: none;
  }

  .modal__footer {
    background: $colorWhite;
    box-shadow: 0px 0px 10px rgba(20, 5, 60, 0.2);
    overflow: visible;
    justify-content: center;
    z-index: 2;
  }

  .modal__footer-left {
    flex: 1;
    margin-right: 10px;
  }

  .modal__footer-right {
    flex: 1;
    margin-left: 10px;
  }
}

.confirm__title {
  margin: 0 20px 20px 20px;
  @include typography(xl, headlines, 500);
  line-height: 26px;
  text-align: center;
  color: $colorSecondaryDark1;
}

.confirm__text {
  margin: 0 0 20px 0;
  @include typography(md, default);
  line-height: 20px;
  text-align: center;
  color: $colorSecondaryLight;
}
</style>
