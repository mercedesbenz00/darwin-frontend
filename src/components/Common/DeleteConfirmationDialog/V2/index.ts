export { default as DeleteConfirmationDialog } from './DeleteConfirmationDialog.vue'
