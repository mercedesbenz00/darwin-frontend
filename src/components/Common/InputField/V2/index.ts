export { default as InputField } from './InputField.vue'
export * from './types'
