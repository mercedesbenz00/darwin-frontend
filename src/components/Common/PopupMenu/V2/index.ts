export { default as PopupMenuV2 } from './PopupMenu.vue'
