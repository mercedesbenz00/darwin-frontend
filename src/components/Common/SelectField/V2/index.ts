export { ChevronIcon } from './assets/icons'

export { default as SelectField } from './SelectField.vue'
