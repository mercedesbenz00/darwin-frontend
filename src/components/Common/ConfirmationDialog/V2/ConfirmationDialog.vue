<template>
  <modal-v2
    :name="name"
    size="medium"
    :width="388"
    transition="pop-out"
    height="auto"
    :min-width="200"
    :max-width="388"
    :classes="['confirm']"
    :click-to-close="clickToClose"
    @before-open="onBeforeOpen"
    @closed="onClosed"
  >
    <modal-content-v2>
      <slot />
      <div class="confirm__title">
        {{ title }}
      </div>
      <div class="confirm__description">
        {{ detail }}
      </div>
      <div class="confirm__footer">
        <custom-button
          class="btn-positive"
          size="medium"
          flair="rounded"
          variant="outline"
          @click.stop.prevent="cancel"
        >
          {{ cancelButtonText }}
        </custom-button>

        <custom-button
          class="btn-negative"
          size="medium"
          flair="rounded"
          :color="confirmButtonColor"
          @click.stop.prevent="confirm"
        >
          {{ buttonText }}
        </custom-button>
      </div>
    </modal-content-v2>
  </modal-v2>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

import { CustomButton } from '@/components/Common/Button/V2'
import ModalV2 from '@/components/Common/Modal/V2/Modal.vue'
import ModalContentV2 from '@/components/Common/Modal/V2/ModalContent.vue'
import { useModal, useStore } from '@/composables'

/**
 * Wrapper around a vue-js-modal used for simple confirm/cancel dialogs.
 *
 * The name property is used to open and close the modal,
 * so multiple modals rendered within the same template need to have unique names.
 */
export default defineComponent({
  name: 'ConfirmationDialog',
  components: {
    CustomButton,
    ModalV2,
    ModalContentV2
  },
  props: {
    title: {
      required: true,
      type: String
    },
    detail: {
      required: true,
      type: String
    },
    buttonText: {
      required: false,
      type: String,
      default: 'Okey'
    },
    confirmButtonColor: {
      required: false,
      type: String,
      default: 'primary'
    },
    cancelButtonText: {
      required: false,
      type: String,
      default: 'Cancel'
    },
    name: {
      required: false,
      type: String,
      default: 'confirm-dialog'
    },
    loading: {
      required: false,
      type: Boolean,
      default: false
    },
    clickToClose: {
      required: false,
      type: Boolean,
      default: true
    }
  },
  setup (props, { emit }) {
    const store = useStore()
    const modal = useModal()

    const onBeforeOpen = (): void => {
      store.dispatch('ui/putBackSidebar')
    }

    const onClosed = (): void => {
      store.dispatch('ui/bringFrontSidebar')
    }

    const show = (): void => {
      modal.show(props.name)
    }

    const close = (): void => {
      modal.hide(props.name)
    }

    const cancel = (): void => {
      emit('canceled')
      close()
    }

    const confirm = (): void => {
      emit('confirmed')
    }

    return {
      onBeforeOpen,
      onClosed,
      show,
      close,
      cancel,
      confirm
    }
  }
})
</script>

<style lang="scss" scoped>
.confirm__title {
  @include typography(xl, inter, 500);
  @include row--center;
  color: $colorContentDefault;

  margin-top: 16px;
  width: 100%;
}

.confirm__description {
  @include typography(lg, inter, 500);
  @include row--center;
  color: $colorContentSecondary;

  margin-top: 8px;
  margin-bottom: 16px;
  width: 100%;
  display: flex;
}

.confirm__footer {
  padding-top: 16px;
  width: 100%;
  display: flex;
  gap: 8px;
  justify-content: space-between;

  .btn-positive, .btn-negative {
    width: 100% !important;
  }
}
</style>
