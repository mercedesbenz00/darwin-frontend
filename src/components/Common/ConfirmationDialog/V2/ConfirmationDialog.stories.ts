import { action } from '@storybook/addon-actions'
import Vue from 'vue'
import Modal from 'vue-js-modal'

import { ConfirmationDialog } from '@/components/Common/ConfirmationDialog/V2'
import store from '@/store'

const stories = {
  title: 'Common/ConfirmationDialog/Versions/V2',
  argTypes: {
    title: { control: 'text' },
    detail: { control: 'text' },
    buttonText: { control: 'text' }
  }
}

export default stories

const buildActions = () => ({
  canceled: action('canceled'),
  confirmed: action('confirmed')
})

type Args = Record<keyof typeof stories.argTypes, any>

export const Default = (args: Args, { argTypes }: any) => ({
  components: { Modal, ConfirmationDialog },
  data: () => ({ actions: buildActions() }),
  props: Object.keys(argTypes),
  mounted () {
    (this as unknown as Vue).$modal.show('confirm-dialog')
  },
  store,
  template: `
    <confirmation-dialog
      v-bind="$props"
      name="confirm-dialog"
      v-on="actions"
    />
  `
})

Default.args = {
  title: 'Are you sure you wish to do this?',
  detail: 'Make sure you are completely sure you want to do this.',
  buttonText: 'Okay'
}
