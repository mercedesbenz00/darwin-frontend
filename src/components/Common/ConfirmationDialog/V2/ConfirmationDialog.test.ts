import { createLocalVue, shallowMount } from '@vue/test-utils'
import VModal from 'vue-js-modal'
import Vuex from 'vuex'

import { createMockTheme } from 'test/unit/components/mocks'
import { stubDirectiveWithAttribute } from 'test/unit/directiveStubs'
import { ConfirmationDialogLayout } from 'test/unit/stubs'

import { ConfirmationDialog } from '@/components/Common/ConfirmationDialog/V2'
import { installCommonComponents } from '@/plugins/components'

const localVue = createLocalVue()
localVue.use(VModal)
localVue.use(Vuex)
localVue.directive('loading', stubDirectiveWithAttribute)
installCommonComponents(localVue)

const mocks = { $theme: createMockTheme() }
const stubs = { ConfirmationDialogLayout }

let propsData: {
  title: string
  detail: string
  buttonText: string
  name?: string
  loading?: boolean
}

const itMatchesSnapshot = () => it('matches snapshot', () => {
  const wrapper = shallowMount(ConfirmationDialog, { localVue, mocks, propsData, stubs })
  expect(wrapper).toMatchSnapshot()
})

describe('with normal props', () => {
  beforeEach(() => {
    propsData = { name: 'test-dialog', title: 'Test', detail: 'Test content', buttonText: 'OK' }
  })

  itMatchesSnapshot()
})
