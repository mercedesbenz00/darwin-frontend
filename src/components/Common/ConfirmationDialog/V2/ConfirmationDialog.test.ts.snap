// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with normal props matches snapshot 1`] = `
<modal-v2-stub size="medium" clicktoclose="true" name="test-dialog" width="388" transition="pop-out" height="auto" min-width="200" max-width="388" classes="confirm">
  <modal-content-v2-stub>
    <div class="confirm__title">
      Test
    </div>
    <div class="confirm__description">
      Test content
    </div>
    <div class="confirm__footer">
      <custom-button-stub variant="outline" color="secondary" size="medium" flair="rounded" tag="button" class="btn-positive">
        Cancel
      </custom-button-stub>
      <custom-button-stub variant="default" color="primary" size="medium" flair="rounded" tag="button" class="btn-negative">
        OK
      </custom-button-stub>
    </div>
  </modal-content-v2-stub>
</modal-v2-stub>
`;
